dali/operators/image/color/equalize.cu

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+
+#include "dali/kernels/dynamic_scratchpad.h"
+#include "dali/kernels/imgproc/color_manipulation/equalize/equalize.h"
+#include "dali/kernels/kernel_manager.h"
+#include "dali/operators/image/color/equalize.h"
+
+namespace dali {
+
+namespace equalize {
+
+class EqualizeGPU : public Equalize<GPUBackend> {
+  using Kernel = kernels::equalize::EqualizeKernelGpu;
+
+ public:
+  explicit EqualizeGPU(const OpSpec &spec) : Equalize<GPUBackend>(spec) {
+    kmgr_.Resize<Kernel>(1);
+  }
+
+ protected:
+  void RunImpl(Workspace &ws) override {
+    const auto &input = ws.Input<GPUBackend>(0);
+    auto &output = ws.Output<GPUBackend>(0);
+    auto input_type = input.type();
+    auto layout = input.GetLayout();
+    DALI_ENFORCE(input_type == type2id<uint8_t>::value,
+                 make_string("Unsupported input type for equalize operator: ", input_type,
+                             ". Expected input type: `uint8_t`."));
+    // enforced by the layouts specified in operator schema
+    assert(layout.size() == 2 || layout.size() == 3);
+    output.SetLayout(layout);
+    kernels::DynamicScratchpad scratchpad({}, AccessOrder(ws.stream()));
+    ctx_.gpu.stream = ws.stream();
+    ctx_.scratchpad = &scratchpad;
+    auto out_view = view<uint8_t>(output);
+    auto in_view = view<const uint8_t>(input);
+    auto out_shape = GetFlattenedShape(out_view.shape);
+    auto in_shape = GetFlattenedShape(in_view.shape);
+    TensorListView<StorageGPU, uint8_t, 2> out_view_flat{out_view.data, out_shape};
+    TensorListView<StorageGPU, const uint8_t, 2> in_view_flat{in_view.data, in_shape};
+    kmgr_.Run<Kernel>(0, ctx_, out_view_flat, in_view_flat);
+  }
+
+  template <int ndim>
+  TensorListShape<2> GetFlattenedShape(TensorListShape<ndim> shape) {
+    if (shape.sample_dim() == 3) {  // has_channels
+      return collapse_dims<2>(shape, {{0, shape.sample_dim() - 1}});
```

Comment:
The layout can be HWC and CHW, doesn't it impact anything here?

Reply:
SequenceOperator handles the latter